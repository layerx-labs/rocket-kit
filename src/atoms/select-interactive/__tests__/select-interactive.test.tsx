import 'jest-styled-components';
import React from 'react';
import { render } from '@testing-library/react';
import { SelectInteractive } from '../../..';

describe('Select Interactive', () => {
  const options = [
    { value: 'chocolate', label: 'Chocolate' },
    { value: 'strawberry', label: 'Strawberry' },
    { value: 'vanilla', label: 'Vanilla' },
    { value: 'cookies', label: 'Cookies and Cream' },
    { value: 'oreo', label: 'Oreo' },
  ];

  it('renders', () => {
    const { asFragment } = render(
      <SelectInteractive
        multi={false}
        search={true}
        placeholder="Awesome Select Component"
        value={options[2]}
        options={options}
        onChange={() => {}}
        onInputChange={() => {}}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
