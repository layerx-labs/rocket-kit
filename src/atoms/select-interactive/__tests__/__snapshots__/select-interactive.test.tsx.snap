// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Interactive renders 1`] = `
<DocumentFragment>
  .c0 .select-interactive {
  z-index: 3;
}

.c0 .select__control {
  border-color: #D9D4ED;
  border-radius: 0.5rem;
  min-height: 3.125rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 .select__control:hover {
  border-color: #C7BFE4;
}

.c0 .select__control--is-focused {
  border-color: #4329A6;
  box-shadow: none;
}

.c0 .select__value-container {
  padding: 0 0.9375rem;
}

.c0 .select__placeholder {
  color: #C7C8D1;
}

.c0 .select__input {
  color: #000000;
}

.c0 .select__single-value {
  color: #000000;
}

.c0 .select__indicator {
  padding: 0 0.9375rem;
}

.c0 .select__indicator:hover {
  color: inherit;
}

.c0 .select__indicator svg {
  width: 1.375rem;
  height: 1.375rem;
}

.c0 .select__indicator svg path {
  fill: #D9D4ED;
}

.c0 .select__indicator-separator {
  background-color: #D9D4ED;
}

.c0 .select__option {
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 .select__option:hover {
  background-color: #C7BFE4;
  color: #000000;
}

.c0 .select__option:hover svg {
  fill: #000000;
}

.c0 .select__option--is-focused {
  background-color: #4329A6;
  color: #ffffff;
}

.c0 .select__option--is-selected {
  background-color: #4329A6 !important;
  color: #ffffff !important;
}

.c0 .select__value-container > div,
.c0 .select__option {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .select__value-container.select__value-container--is-multi > div {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c0 .select__value-container.select__value-container--is-multi > div .select__multi-value__remove {
  padding-right: 0;
}

.c0 .select__value-container svg,
.c0 .select__option svg,
.c0 .select__value-container img,
.c0 .select__option img {
  margin: 0 0.3125rem 0 0;
  width: 1.25rem;
  height: auto;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 .select__option--is-selected svg {
  fill: #ffffff !important;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 .select__option--is-focused svg {
  fill: #ffffff;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c0 .select__option--is-disabled {
  pointer-events: none;
}

.c1 {
  grid-area: 1 / 1 / 2 / 3;
  margin-left: 0.125rem;
  margin-right: 0.125rem;
  max-width: 100%;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  box-sizing: border-box;
}

<div
    class="c0"
  >
    <div
      class="select-interactive css-b62m3t-container"
    >
      <span
        class="css-1f43avz-a11yText-A11yText"
        id="react-select-2-live-region"
      />
      <span
        aria-atomic="false"
        aria-live="polite"
        aria-relevant="additions text"
        class="css-1f43avz-a11yText-A11yText"
      />
      <div
        class="select__control css-13cymwt-control"
      >
        <div
          class="select__value-container select__value-container--has-value css-1fdsijx-ValueContainer"
        >
          <div
            class="c1"
          >
            Vanilla
          </div>
          <div
            class="select__input-container css-qbdosj-Input"
            data-value=""
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-haspopup="true"
              autocapitalize="none"
              autocomplete="off"
              autocorrect="off"
              class="select__input"
              id="react-select-2-input"
              role="combobox"
              spellcheck="false"
              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
              tabindex="0"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="select__indicators css-1hb7zxy-IndicatorsContainer"
        >
          <div
            aria-hidden="true"
            class="select__indicator select__clear-indicator css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
              />
            </svg>
          </div>
          <span
            class="select__indicator-separator css-1u9des2-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            class="select__indicator select__dropdown-indicator css-1xc3v61-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-tj5bde-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
